### PR TITLE
docs: eslint flat config

### PR DESCRIPTION
**Pull request by devjiwonchoi**
This PR replaced eslint-related docs code examples from `.eslintrc.json` with `eslint.config.mjs`.

Closes NDX-373

### CODE REVIEW

File: docs/01-app/03-api-reference/05-config/03-eslint.mdx [hunk 1]
```diff
@@ -72,15 +72,25 @@ next lint --dir pages --dir utils --file bar.js
...
-    "next": {
-      "rootDir": "packages/my-app/"
-    }
-  }
-}
+```mjs filename="eslint.config.mjs"
+import { FlatCompat } from '@eslint/eslintrc'
+
+const compat = new FlatCompat({
+  baseDirectory: import.meta.dirname,
```
Comment (ijjk) [inline docs/01-app/03-api-reference/05-config/03-eslint.mdx:79]: Maybe we add a comment mentioning Node.js version requirement for this

**Review by ijjk — APPROVED** [review]

**Review by samcx — APPROVED** [review]
<picture data-single-emoji=":lfgtm-intense:" title=":lfgtm-intense:">:lfgtm-intense:</picture>

**Comment by nikosantis** [comment]
Using eslint 9 I'm getting this error:
"TypeError: Error while loading rule '@typescript-eslint/no-unused-expressions': Cannot read properties of undefined (reading 'allowShortCircuit')"
`typescript-eslint` was updated to fix this. Check:

https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.15.0 
cc: @devjiwonchoi @ijjk

**Comment by devjiwonchoi** [comment]
@nikosantis Thanks for the report! We have a dependency range with `^8.0.0`, could you try uninstall & reinstall `eslint-config-next` to bump the deps?

**Comment by robcaldecott** [comment]
I had the same error as @nikosantis when upgrading to Next.js 15.1.0 and eslint 9. `npm ls typescript-eslint` shows the package is missing. I had to install it manually to make this work.

Here's my `eslint.config.mjs` config:

```js
import { dirname } from "path";
import { fileURLToPath } from "url";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const compat = new FlatCompat({
  baseDirectory: __dirname,
});

const eslintConfig = [
  ...compat.extends("next/core-web-vitals", "next/typescript"),
];

export default eslintConfig;
```

And the relevant packages:

```
"next": "^15.1.0",
"react": "^19.0.0",
"react-dom": "^19.0.0",
"@eslint/eslintrc": "^3.2.0",
"eslint-config-next": "^15.1.0",
"typescript": "^5.7.2",
```

So I can't see `eslint-config-next` is pulling in all the TS packages it needs.

**Comment by nikosantis** [comment]
> @nikosantis Thanks for the report! We have a dependency range with `^8.0.0`, could you try uninstall & reinstall `eslint-config-next` to bump the deps?

Removed the lock json file (pnpm) and node modules and install all again works @devjiwonchoi 
Check this @robcaldecott !

**Comment by robcaldecott** [comment]
Yep, that worked @nikosantis I should have thought of that, thanks.

**Pull request merged**